activity/src/test/java/com/amazonaws/stepfunctions/cloudformation/activity/DeleteHandlerTest.java

```diff
@@ -80,4 +95,34 @@ public void testUnknownException() {
         assertThat(response.getMessage()).isEqualTo(Constants.INTERNAL_FAILURE_MESSAGE);
     }
 
+    @Test
+    public void testResourceArnIsNull_returnsNotFound() {
+        request.setDesiredResourceState(ResourceModel.builder()
+                .name(ACTIVITY_NAME)
+                .build());
+
+        final ProgressEvent<ResourceModel, CallbackContext> response
+                = handler.handleRequest(proxy, request, null, logger);
+
+        assertThat(response).isNotNull();
+        assertThat(response.getStatus()).isEqualTo(OperationStatus.FAILED);
+        assertThat(response.getErrorCode()).isEqualTo(HandlerErrorCode.NotFound);
```

Comment:
Do we want to assert on the exception message as well? because the notFound error code mapped to 3 conditions and would be nice we can validate that as well

Reply:
Added assertions on message as well!